### PR TITLE
Feature/incoming number

### PR DESCRIPTION
**Pull request by DragonRaider5**
Hey,
regarding to #2 , I added the ability to get the phone number of an incoming call. Unfortunately this is only supported under Android, it's not possible in iOS due to privacy restrictions I guess.
It is beeing passed to the callback as a 2nd argument.

### CODE REVIEW

File: android/src/main/java/com/pritesh/calldetection/CallDetectionManagerModule.java [hunk 1]
```diff
@@ -120,13 +120,13 @@ public void phoneCallStateUpdated(int state) {
...
                 //Device call state: Off-hook. At least one call exists that is dialing, active, or on hold, and no calls are ringing or waiting.
                 wasAppInOffHook = true;
-                jsModule.callStateUpdated("Offhook");
+                jsModule.callStateUpdated("Offhook", null);
                 break;
             //Incoming
             case TelephonyManager.CALL_STATE_RINGING:
                 // Device call state: Ringing. A new call arrived and is ringing or waiting. In the latter case, another call is already active.
-                jsModule.callStateUpdated("Incoming");
+                jsModule.callStateUpdated("Incoming", incomingNumber);
```
Comment (priteshrnandgaonkar) [inline android/src/main/java/com/pritesh/calldetection/CallDetectionManagerModule.java]: Instead `incomingNumber` you should have `phoneNumber`

File: index.js [hunk 2]
```diff
@@ -10,19 +15,35 @@ const NativeCallDetectorAndroid = NativeModules.CallDetectionManagerAndroid
...
         this.callback = callback
         if (Platform.OS === 'ios') {
             NativeCallDetector && NativeCallDetector.startListener()
             this.subscription = new NativeEventEmitter(NativeCallDetector)
             this.subscription.addListener('PhoneCallStateUpdate', callback);
         }
         else {
-            NativeCallDetectorAndroid && NativeCallDetectorAndroid.startListener()
+            if(NativeCallDetectorAndroid) {
+              requestPermissionsAndroid(permissionMessage)
```
Comment (priteshrnandgaonkar) [inline index.js]: Don't make permission mandatory. As few users might not be interested in reading the incoming phone number, they might be interested in just detecting the incoming call. Please make this new feature optional and not a mandatory task. Just take the argument from the constructor which will indicate if they want to read incoming number, based on that value you can decide if you want to show permission or not.

File: CallDetectionExample/android/local.properties [hunk 3]
```diff
@@ -1,11 +0,0 @@
-## This file is automatically generated by Android Studio.
```
[the remark this answers is not in the document]
Comment (DragonRaider5) [reply CallDetectionExample/android/local.properties]: You're welcome :)

File: android/src/main/java/com/pritesh/calldetection/CallDetectionManagerModule.java (hunk 1 above)
Comment (DragonRaider5) [reply android/src/main/java/com/pritesh/calldetection/CallDetectionManagerModule.java]: Oh right, thanks!

File: index.js (hunk 2 above)
Comment (DragonRaider5) [reply index.js]: Done.

**Comment by priteshrnandgaonkar** [comment]
Thanks for the feature

**Pull request merged**